RNGNewAuraNotifier/Core/Config/AppConfig.cs

```diff
@@ -7,120 +7,132 @@ namespace RNGNewAuraNotifier.Core.Config;
 /// </summary>
 internal class AppConfig
 {
-    /// <summary>
-    /// 設定ファイルのパス
-    /// </summary>
-    /// <remarks>アプリケーションの実行ディレクトリに config.json という名前で保存される</remarks>
-    /// <example>config.json</example>
-    private static readonly string _configFilePath = Path.Combine(Environment.CurrentDirectory, "config.json");
+    private const string PathFileName = "config.path";
+    private const string ConfigFileName = "config.json";
+
+    private static string _configDir = GetConfigDirectoryPath();
+    private static ConfigData _instance = new();
+    private static readonly Lock _lock = new();
+    private static bool _isLoaded = false;
 
     /// <summary>
-    /// 設定データ
+    /// 設定が再読み込みされたときに発生するイベント
     /// </summary>
-    private static ConfigData _config = new();
+    public static event Action<ConfigData>? ConfigReloaded;
```

Comment:
イベントを示すものなので、On~~~ のほうがわかりやすいかなと思っているのですがどうでしょう。ほかでもOnDetectedとかにしてます

```suggestion
    public static event Action<ConfigData>? OnConfigReloaded;
```